lib/Service/Appointments/BookingCalendarWriter.php

```diff
@@ -236,4 +253,89 @@ public function write(AppointmentConfig $config, DateTimeImmutable $start, strin
 		}
 		return $vcalendar->serialize();
 	}
+
+	/**
+	 * Returns a VTIMEZONE component for a Olson timezone identifier
+	 * with daylight transitions covering the given date range.
+	 *
+	 * @link https://gist.github.com/thomascube/47ff7d530244c669825736b10877a200
+	 *
+	 * @param string $timezone Timezone
+	 * @param integer $from Unix timestamp with first date/time in this timezone
+	 * @param integer $to Unix timestap with last date/time in this timezone
+	 *
+	 * @return null|Component A Sabre\VObject\Component object representing a VTIMEZONE definition
+	 *               or null if no timezone information is available
+	 */
+	private function generateVtimezone($timezone, $from, $to): ?Component {
+		try {
+			$tz = new \DateTimeZone($timezone);
+		}
+		catch (\Exception $e) {
+			return null;
+		}
+
+		// get all transitions for one year back/ahead
+		$year = 86400 * 360;
+		$transitions = $tz->getTransitions($from - $year, $to + $year);
+
+		$vcalendar = new VCalendar();
+		$vtimezone = $vcalendar->createComponent('VTIMEZONE');
+		$vtimezone->TZID = $timezone;
+
+		$standard = $daylightStart = null;
+		foreach ($transitions as $i => $trans) {
+			$component = null;
+
+			// skip the first entry...
+			if ($i == 0) {
```

Comment:
```suggestion
			if ($i === 0) {
```

it's copied code but let's bring it up to our standards